Branch-SDK/src/io/branch/referral/util/CommerceEvent.java

```diff
@@ -137,6 +131,7 @@ public JSONObject getCommerceJSONObject() {
     }
 
     public List<JSONObject> getProducts() {
+        if (this.products == null) return null;
```

Comment:
should we return null here? Can we return an empty List in order to prevent any accidental  NPE